src/unix/fs.c

```diff
@@ -48,7 +48,8 @@
 #if defined(__DragonFly__)  ||                                            \
     defined(__FreeBSD__)    ||                                            \
     defined(__OpenBSD__)    ||                                            \
-    defined(__NetBSD__)
+    defined(__NetBSD__)     ||                                            \
+    defined(__linux__)
 # define HAVE_PREADV 1
 #else
 # define HAVE_PREADV 0
```

Comment:
RHEL / CentOS 5.  They ship with 2.6.18, have a sizable installed user base and are supported until 2017.